### PR TITLE
#143358935 User creates and starts game

### PR DESCRIPTION
**Pull request by andela-onnenanya**
#### What does this PR do?
Allows user to click on a button to start a game
#### Description of Task to be completed?
Users should be able to create/start a new game
#### How should this be manually tested?
Click on the "Start Game" button close to logout button.
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
Users should be able to create/start a new game
#### Screenshots (if appropriate)
N/A
#### Questions:
N/A

### CODE REVIEW

File: app/controllers/game.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ * start game if request is authenticated
+ * @return {void}
+ * @param {string} req - request from client end.
+ * @param {string} res - response send to client end.
+ */
+
+exports.start = (req, res) => {
+  const newGame = new Game();
+  // game_id should use _id
+  // this is just a sample data, should be updated for real user
```
Comment (emmyakin) [inline app/controllers/game.js:14]: no need to use dummy data, you can construct the game data fro the request object
e.g 
```
const gameOnwer_id = req.body.gameOwnerId;
and the rest.
```

File: app/controllers/index.js [hunk 2]
```diff
@@ -1,23 +1,27 @@
 /**
  * Module dependencies.
  */
-var mongoose = require('mongoose'),
-    async = require('async'),
-    _ = require('underscore');
+// const mongoose = require('mongoose');
```
Comment (emmyakin) [inline app/controllers/index.js]: delete the lines rather than comment if we no longer need them

File: app/models/game.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+/**
+ * Module dependencies.
+ */
+const mongoose = require('mongoose');
+const Schema = mongoose.Schema;
+/**
+ * Game Schema
+ */
+const Game = new Schema({
+  game_id: String,
```
Comment (emmyakin) [inline app/models/game.js]: maybe we should use camelCase for the schema fields

File: public/js/controllers/index.js [hunk 4]
```diff
@@ -6,6 +7,37 @@ angular.module('mean.system')
...
       $scope.errorMsg = '';
 
+      $scope.startGame = () => {
+        swal({
+          title: 'Are you sure?',
+          text: 'You want to start the game now?',
+          type: 'success',
+          showCancelButton: true,
+          confirmButtonColor: '#DD6B55',
+          confirmButtonText: 'Yes Am Ready',
```
Comment (emmyakin) [inline public/js/controllers/index.js]: Yes / No

File: public/js/controllers/index.js [hunk 5]
```diff
@@ -6,6 +7,37 @@ angular.module('mean.system')
       $scope.global = Global;
       $scope.errorMsg = '';
 
+      $scope.startGame = () => {
+        swal({
+          title: 'Are you sure?',
+          text: 'You want to start the game now?',
```
Comment (emmyakin) [inline public/js/controllers/index.js]: let's use "Start a new Game?"

File: public/js/controllers/index.js [hunk 6]
```diff
@@ -6,6 +7,37 @@ angular.module('mean.system')
...
+          showCancelButton: true,
+          confirmButtonColor: '#DD6B55',
+          confirmButtonText: 'Yes Am Ready',
+          cancelButtonText: 'No, Not Now',
+          closeOnConfirm: true,
+          closeOnCancel: true
+        },
+        (isConfirm) => {
+          if (isConfirm) {
+            $http.post('/api/games/:id/start').then(
```
Comment (emmyakin) [inline public/js/controllers/index.js]: can you move this into the user service, so we don't have to inject $http here

**Review by emmyakin — CHANGES_REQUESTED** [review]

File: public/js/controllers/index.js (hunk 6 above)
Comment (andela-onnenanya) [reply public/js/controllers/index.js]: Ok, working on all the reviews.

File: public/js/controllers/index.js (hunk 5 above)
Comment (andela-onnenanya) [reply public/js/controllers/index.js]: For line 12 or 13

File: public/js/controllers/index.js [hunk 7]
```diff
@@ -1,10 +1,41 @@
...
+            const data = {
+              gameOwnerId: user.id,
+              players: [user.id]
+            };
+            userService.startGame(data).then(({ data }) => {
+              Global.setCurrentGameId(data._id);
+              $location.path('/app');
+            });
+          } else {
+            swal('Cancelled', 'You are off! Shitty you!!!', 'error');
```
Comment (emmyakin) [inline public/js/controllers/index.js:35]: uhm , can you use a proper message like game, cancelled.

And add (error) => {} callback where you check the status code and redirect the user to login page or show a server error message.

**Review by emmyakin — CHANGES_REQUESTED** [review]

**Review by emmyakin — APPROVED** [review]

**Pull request merged**